cdk/stack.py

```diff
@@ -98,6 +102,76 @@ def __init__(
             description="Name of table used to store ingestions",
         )
 
+        env_secret = self.build_env_secret(config.stage, env)
+        secret_arn: str = env_secret.secret_arn
+
+        self.build_oidc(
+            oidc_provider_arn=config.oidc_provider_arn,
+            oidc_repo_id=config.oidc_repo_id,
+            secret_arn=secret_arn,
+            stage=config.stage,
+        )
+
+    def build_oidc(
+        self, oidc_provider_arn: str, oidc_repo_id: str, secret_arn: str, stage: str
+    ):
+        if oidc_provider_arn:
+            # Create an IAM OIDC provider for the specified provider ARN
+            oidc_provider = iam.OpenIdConnectProvider.from_open_id_connect_provider_arn(
+                self, "OIDCProvider", oidc_provider_arn
+            )
+            # create IAM role for provider access from specified repo
+            # the role should allow a github action in that repo
+            # to deploy resources (TODO) and read a secret
+            oidc_role = iam.Role(
+                self,
+                f"stac-ingestor-oidc-role-{stage}",
+                assumed_by=iam.WebIdentityPrincipal(
+                    oidc_provider.open_id_connect_provider_arn,
+                    conditions={
+                        "StringEquals": {
+                            f"{oidc_provider.open_id_connect_provider_issuer}:sub": f"repo:{oidc_repo_id}"  # noqa E501
+                        }
+                    },
+                ),
+            )
+            oidc_role.add_to_policy(
+                iam.PolicyStatement(
+                    actions=["sts:AssumeRoleWithWebIdentity"],
+                    resources=[oidc_provider_arn],
+                )
+            )
+            # Create an IAM policy statement that allows getting the secret value
+            get_secret_statement = iam.PolicyStatement(
+                effect=iam.Effect.ALLOW,
+                actions=["secretsmanager:GetSecretValue"],
+                resources=[secret_arn],
+            )
+
+            oidc_policy = iam.Policy(
+                self,
+                f"stac-ingestor-oidc-policy-{stage}",
+                policy_name=f"stac-ingestor-oidc-policy-{stage}",
+                roles=[oidc_role],
+                statements=[get_secret_statement],
+            )
+            return oidc_role, oidc_policy, oidc_provider
```

Comment:
are these returns being used anywhere?

Reply:
No, they just stop the linter from complaining